### PR TITLE
Prevent multiple question submissions

### PR DESCRIPTION
**Pull request by spleshka**

### CODE REVIEW

File: src/Quiz.php [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+   * @param EntityInterface $node
+   *   Lesson node.
+   *
+   * @return array
+   *   An array containing current node and referenced course.
+   */
+  public function getPageData(NodeInterface|EntityInterface $node)
+  {
+    $data = parent::getPageData($node);
+    return $this->_anu_lms_get_quiz_submission_data($node, $data);
```
Comment (spleshka) [inline src/Quiz.php]: you don't need to start with underscore the protected methods. Move over, the methods use lowerCamelCase naming convension.

File: src/Lesson.php [hunk 2]
```diff
@@ -32,7 +34,8 @@ class Lesson {
    * @param \Drupal\anu_lms\Normalizer $normalizer
    *   The normalizer.
    */
-  public function __construct(EntityTypeManagerInterface $entity_type_manager, Normalizer $normalizer) {
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, Normalizer $normalizer)
```
Comment (spleshka) [inline src/Lesson.php]: This change is not compliant with Drupal coding standards, see https://www.drupal.org/docs/develop/standards/object-oriented-code for details.

File: src/Quiz.php [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            $correct_answers++;
+          }
+        }
+        if (!empty($results)) {
+          $data['results'] = $results;
+          $data['correct_answers'] = $correct_answers;
+        }
+      }
+    }
+    return $data;
```
Comment (spleshka) [inline src/Quiz.php:84]: The variable was passed here by reference, so either no need to return the value (it has been modified globally) or don't pass it by reference.

File: src/Quiz.php [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  /**
+   * Returns data for Quiz submissions.
+   *
+   * @param NodeInterface $node
+   *   Quiz node
+   *
+   * @param array $data
+   *   Data to be returned
+   */
+  protected function _anu_lms_get_quiz_submission_data(NodeInterface $node, array &$data)
```
Comment (spleshka) [inline src/Quiz.php]: Naming for such methods should just be "getSubmissionData()" instead, according to the Drupal naming conventions.

File: src/Quiz.php [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+  /**
+   * Returns data for Quiz submissions.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $node
+   *   Quiz node.
+   * @param array $data
+   *   Data to be returned.
+   */
+  protected function anuLmsGetQuizSubmissionData(EntityInterface $node, array &$data) {
```
Comment (spleshka) [inline src/Quiz.php]: no need to prefix methods with anuLms @esbaddeley

File: src/Quiz.php (hunk 3 above)
Comment (spleshka) [reply src/Quiz.php:84]: @esbaddeley can you address this one as well? Thanks.

File: src/Lesson.php (hunk 2 above)
Comment (esbaddeley) [reply src/Lesson.php]: done

File: src/Quiz.php (hunk 4 above)
Comment (esbaddeley) [reply src/Quiz.php]: done

File: src/Quiz.php (hunk 5 above)
Comment (esbaddeley) [reply src/Quiz.php]: changed

File: src/Quiz.php (hunk 3 above)
Comment (esbaddeley) [reply src/Quiz.php:84]: passing by value not reference and returning

**Pull request merged**